src/palladio/SOPAssign.cpp

```diff
@@ -398,6 +400,119 @@ bool evaluateDefaultRuleAttributes(SOPAssign* node, const GU_Detail* detail, Sha
 	return true;
 }
 
+double getDefaultBool(const SOPAssign::CGAAttributeValueType& defaultValue) {
+	if (std::holds_alternative<bool>(defaultValue))
+		return std::get<bool>(defaultValue);
+	return false;
+}
+
+double getDefaultFloat(const SOPAssign::CGAAttributeValueType& defaultValue) {
+	if (std::holds_alternative<double>(defaultValue))
+		return std::get<double>(defaultValue);
+	return 0;
+}
+
+std::wstring getDefaultString(const SOPAssign::CGAAttributeValueType& defaultValue) {
+	if (std::holds_alternative<std::wstring>(defaultValue))
+		return std::get<std::wstring>(defaultValue);
+	return {};
+}
+
+std::wstring getDescription(const AnnotationParsing::TraitParameterMap& traitParmMap) {
+	const auto& descriptionIt = traitParmMap.find(AnnotationParsing::AttributeTrait::DESCRIPTION);
+	if (descriptionIt != traitParmMap.end())
+		return std::get<std::wstring>(descriptionIt->second);
```

Comment:
this get could (theoretically) throw as we do not check for the variant type

there are some more cases below, maybe wrap in try/catch or use std::get_if?

Reply:
using std::holds_alternative in the if above, like in the other examples